feature/experimental/p4rt/ate_tests/base_p4rt/base_p4rt_test.go

```diff
@@ -0,0 +1,422 @@
+			ElectionId: &p4_v1.Uint128{High: uint64(0), Low: electionId},
+			Action:     p4_v1.SetForwardingPipelineConfigRequest_VERIFY_AND_COMMIT,
+			Config: &p4_v1.ForwardingPipelineConfig{
+				P4Info: &p4Info,
+				Cookie: &p4_v1.ForwardingPipelineConfig_Cookie{
+					Cookie: 159,
+				},
+			},
+		}); err != nil {
+			return errors.New("Errors seen when sending SetForwardingPipelineConfig.")
+		}
+	}
+
+	// Receive GetForwardingPipelineConfig
+	for index, client := range clients {
+		resp, err := client.GetForwardingPipelineConfig(&p4_v1.GetForwardingPipelineConfigRequest{
+			DeviceId:     deviceid_list[index],
+			ResponseType: p4_v1.GetForwardingPipelineConfigRequest_P4INFO_AND_COOKIE,
+		})
+		if err != nil {
+			return errors.New("Errors seen when sending SetForwardingPipelineConfig.")
+		}
+		// Compare P4Info from GetForwardingPipelineConfig and SetForwardingPipelineConfig
+		getp4Info, _ := json.Marshal(resp.Config.P4Info)
+		fmt.Println(string(getp4Info))
+		if string(setp4Info) == string(getp4Info) {
+			log.Info("P4info matches fine \n")
+		} else {
+			log.Warningf("P4info does not match \n")
+		}
+	}
+	return nil
+}
+
+// Function to compare and check if the expected table is present in RPC ReadResponse
+func verifyReadReceiveMatch(expected_update []*p4_v1.Update, received_entry *p4_v1.ReadResponse) error {
+
+	matches := 0
+	for i, table := range received_entry.Entities {
+		if table.Entity == expected_update[i].Entity.Entity {
+			matches += 1
+		}
+	}
+	if matches == len(expected_update) {
+		log.Infof("Table match succesful")
+		return nil
+	} else {
+		log.Warningf("Table match failed")
+		return errors.New("match unsuccesful")
+	}
+}
+
+// TestP4rtConnect connects to the P4Runtime server over grpc
+// It then calls setupP4RTClient which sets the arbitration request and sends SetForwardingPipelineConfig with P4Info
+func TestP4rtConnect(t *testing.T) {
+
+	dut := ondatra.DUT(t, "dut")
+	ctx := context.Background()
+	ate := ondatra.ATE(t, "ate")
+
+	// configure DUT with P4RT node-id and ids on different FAPs
+	configureDUT(t, dut, []attrs.Attributes{dutPort1, dutPort2}, []string{*comp1name, *comp2name}, []uint64{deviceId1, deviceId2}, []uint32{portId1, portId2})
+
+	top := configureATE(t, ate)
+
+	// Setup two different clients for different FAPs
+	client1 := p4rt_client.P4RTClient{}
+	if err := client1.P4rtClientSet(dut.RawAPIs().P4RT(t)); err != nil {
+		t.Fatalf("Could not initialize p4rt client: %v", err)
+	}
+
+	client2 := p4rt_client.P4RTClient{}
+	if err := client2.P4rtClientSet(dut.RawAPIs().P4RT(t)); err != nil {
+		t.Fatalf("Could not initialize p4rt client: %v", err)
+	}
+
+	args := &testArgs{
+		ctx:     ctx,
+		client1: &client1,
+		client2: &client2,
+		dut:     dut,
+		ate:     ate,
+		top:     top,
+	}
+
+	if err := setupP4RTClient(ctx, args); err != nil {
+		t.Fatalf("Could not setup p4rt client: %v", err)
+	}
+
+	// Get Capbilities
+	_, err := args.client1.Capabilities(&p4_v1.CapabilitiesRequest{})
```

Comment:
Based on the procedure, Capability RPC is not required. I think this can be removed. @hariramnat what do you think?

Reply:
Yes @mingyangcisco , i agree. I will remove these lines in my next commit.

Reply:
This is taken care of

Reply:
thanks @rsindhujha